woocommerce/woocommerce-gateway-stripe
# Update messaging for inactive payment methods

Fixes #3188

## Changes proposed in this Pull Request:

- Remove the generic "Account needs activation" notice
- Display a pill in each payment method row indicating whether it needs activation
- Use a different wording when the payment method is inactive and when the activation request is still pending

| Before | After |
|--------|--------|
| <img width="700" alt="image" src="https://github.com/user-attachments/assets/ecd6004f-189f-4226-b93f-0d23b90d983b"> | <img width="700" alt="image" src="https://github.com/user-attachments/assets/7ce5055a-94d2-4142-a7a7-3ecbd14e1bc6"> |

## Testing instructions

1. Go to the [Payment methods settings page](https://dashboard.stripe.com/settings/payment_methods) in the Stripe dashboard
2. Turn off a few payment methods
  <img width="700" alt="image" src="https://github.com/user-attachments/assets/d529fd3f-c83d-4b9a-9762-9266cea78fc9">

3. Go to the Payment methods settings in the Stripe extension, at `/wp-admin/admin.php?page=wc-settings&tab=checkout&section=stripe&panel=methods`
4. Confirm that:
  - The "Activate Stripe account" notice is no longer displayed at the top of the tab
  - The payment methods that are turned off now have a "Requires activation" tab
  - A modal is displayed when hovering over the "Requires activation" pill

I didn't find a way to have the "Pending activation" status on test mode. I mocked the status in the code:
- Insert a new line [here](https://github.com/woocommerce/woocommerce-gateway-stripe/pull/3301/files#diff-4618ccc3863398f61ae7c2308e7bf82edeac578ff49670b78f133005f106086cL26) and add `capabilities.ideal_payments = 'pending';`
- Run `npm run build:webpack`
- Go to the Payment methods settings in the Stripe extension, at `/wp-admin/admin.php?page=wc-settings&tab=checkout&section=stripe&panel=methods`
- For iDEAL, confirm that i thas a "Pending approval" pill and a tooltip is displayed when you hover over it
    <img width="700" alt="image" src="https://github.com/user-attachments/assets/4704b79f-b18c-449f-9525-84ef309ed018">



---


**Post merge**
